fawltydeps/extract_deps/environment_yml_parser.py

```diff
@@ -64,7 +64,15 @@ def parse_environment_yml_deps(
                 yield parse_item(dep_item, source)
             except ValueError as e:
                 # InvalidCondaRequirement or packaging.requirements.InvalidRequirement
-                logger.error(f"{error_msg} {e}")
+                # Skip editable self-dependencies silently:
+                if dep_item.removesuffix("/") in {
+                    "-e .",
+                    "--editable .",
+                    "--editable=.",
```

Comment:
Is `.` the only value to skip here? Could there be other values that need to be skipped, like `nested/lib` or such ?

Reply:
For now, I think we only want to work around the special case of direct (editable) self-dependency.

AFAICS, this is by far the most common case of this pattern, and I'm not 100% sure if a nested path dependency (editable or not) can be safely ignored...

Reply:
...like, in your `nested/liib` example: Presumably this dependency declaration would make `import lib` available in your Python environment, and I suspect it _should_ be treated by FD as a 3rd-party import?